### PR TITLE
preprocessing step for render-helm-chart private repo tests

### PR DESCRIPTION
**Pull request by natasha41575**
Related to https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/875

Writes the Secrets to the necessary files as a preprocessing step. I'm doing this separately from https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/875 because PRs from forks cannot access Github Secrets. So this needs to go in first, then https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/875 can use it.

### CODE REVIEW

**Comment by natasha41575** [comment]
cc @mengqiy

File: scripts/verify-docs.py
Comment (mengqiy) [inline scripts/verify-docs.py:38]: Please don't forget to remove this change in the other PR.

File: examples/render-helm-chart-kustomize-private-git/credentials.env
Comment (mengqiy) [inline examples/render-helm-chart-kustomize-private-git/credentials.env:2]: Do we need the placeholders here?

File: .github/workflows/ci.yaml
Comment (mengqiy) [inline .github/workflows/ci.yaml:109]: If a contributor run the tests locally, will all tests pass?

File: scripts/verify-docs.py
Comment (natasha41575) [reply scripts/verify-docs.py:38]: Left a comment to remind myself https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/875#issuecomment-1149289269

File: examples/render-helm-chart-kustomize-private-git/credentials.env
Comment (natasha41575) [reply examples/render-helm-chart-kustomize-private-git/credentials.env:2]: Not needed, but thought it might be helpful for us to remember where and how the credentials are stored and generated. Do you prefer we don't include these files?

File: .github/workflows/ci.yaml
Comment (natasha41575) [reply .github/workflows/ci.yaml:109]: No. Is there a good way to mark these directories as unable to be tested locally? 

In kustomize, we gate such tests with an environment variable that we set in CI. I could do something similar here if desired.

Comment (mengqiy) [reply .github/workflows/ci.yaml:109]: > Is there a good way to mark these directories as unable to be tested locally?

The test harness doesn't support it now. Currently we only have `skip`: https://github.com/GoogleContainerTools/kpt/blob/main/pkg/test/runner/README.md

Here is one option: You can add another config field maybe called `skipLocal` in the test harness in the kpt repo. When a developer test is locally, these tests will be skipped. When tests is run in the GH Action, we can use another env var to make all tests (including the ones with `skipLocal: true`) are tested.

If you have better solutions, please feel free to use.

File: examples/render-helm-chart-kustomize-private-git/credentials.env
Comment (mengqiy) [reply examples/render-helm-chart-kustomize-private-git/credentials.env:2]: No need to delete it. But I'd suggest adding more comments to make it clear that the values will be populated by the CI and it doesn't work locally.

File: .github/workflows/ci.yaml
Comment (natasha41575) [reply .github/workflows/ci.yaml:109]: Thanks! Yeah that's what I was thinking too. I created a PR to add this option in the kpt repo: https://github.com/GoogleContainerTools/kpt/pull/3295

File: examples/render-helm-chart-kustomize-private-git/credentials.env
Comment (natasha41575) [reply examples/render-helm-chart-kustomize-private-git/credentials.env:3]: Good idea. Added more details to the comment!

**Comment by natasha41575** [comment]
@mengqiy is this PR blocked by anything?

File: examples/render-helm-chart-kustomize-private-oci/password
Comment (yuwenma) [inline examples/render-helm-chart-kustomize-private-oci/password:2]: is this the secret yaml resource or user/password? I'd suggest give an example to show the required format, but using fake data

**Review by yuwenma — APPROVED** [review]
one minor comment. Defer to @mengqiy to approve

**Pull request merged**
